File: src/euler/problems/p000007.js
var i = 1;
var p = 2;
for (;i <= 10001;) {
	p++;	
	if(isPrime(p)) {
		i++;
	}
}

console.log(p)


function isPrime(n){
	for(let i = 2;i<Math.sqrt(n);i++) {
		if (n % i == 0) return false;
	}
	return true;
}

File: src/euler/problems/p000007.go
package main

import (
	"fmt"
	"math"
)

func main(){
	prime := uint64(2)
	for i := 1;i <= 10001; {
		
		if isPrime(prime) {
			i++
		}
		prime ++
		
	}	
	fmt.Println(prime - 1)
}


func isPrime(n uint64) bool{
	for i:=uint64(2);i <= uint64(math.Sqrt(float64(n)));i++{
		if n % i == 0 {
			return false
		}
	}
	return true
}


